From 1411f79dbca8d5101890457ea812ffb0a1e5ebeb Mon Sep 17 00:00:00 2001
From: Gregory Amerson <gregory.amerson@liferay.com>
Date: Fri, 13 Feb 2015 00:01:59 +0800
Subject: [PATCH] LPS-52961 add support for dylib on mac and fix win32 build

---
 Makefile | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 885db5c..7be230d 100644
--- a/Makefile
+++ b/Makefile
@@ -65,7 +65,7 @@ ifeq ($(UNAME),Darwin)
 endif
 
 ifneq ($(BUILD),shared)
-	BUILD = static
+	BUILD = shared
 endif
 
 ifeq (,$(PREFIX))
@@ -135,7 +135,7 @@ ifeq (MinGW,$(UNAME))
 	ifeq (shared,$(BUILD))
 		CFLAGS    += -D ADD_EXPORTS 
 		CXXFLAGS  += -D ADD_EXPORTS 
-		LIBRARIES += lib/libsass.dll
+		LIBRARIES = lib/sass.dll
 		RESOURCES += res/resource.rc
 	endif
 else
@@ -144,6 +144,10 @@ else
 	LDFLAGS  += -fPIC
 endif
 
+ifeq ($(UNAME),Darwin)
+	LIBRARIES = lib/libsass.dylib
+endif
+
 OBJECTS = $(SOURCES:.cpp=.o)
 COBJECTS = $(CSOURCES:.c=.o)
 RCOBJECTS = $(RESOURCES:.rc=.o)
@@ -171,13 +175,17 @@ lib/libsass.a: $(COBJECTS) $(OBJECTS)
 	$(MKDIR) lib
 	$(AR) rcvs $@ $(COBJECTS) $(OBJECTS)
 
+lib/libsass.dylib: $(COBJECTS) $(OBJECTS)
+	$(MKDIR) lib
+	$(CXX) -dynamiclib $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) $(LDLIBS)
+
 lib/libsass.so: $(COBJECTS) $(OBJECTS)
 	$(MKDIR) lib
 	$(CXX) -shared $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) $(LDLIBS)
 
-lib/libsass.dll: $(COBJECTS) $(OBJECTS) $(RCOBJECTS)
+lib/sass.dll: $(COBJECTS) $(OBJECTS)
 	$(MKDIR) lib
-	$(CXX) -shared $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) $(RCOBJECTS) $(LDLIBS) -s -Wl,--subsystem,windows,--out-implib,lib/libsass.a
+	$(CXX) -shared $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) -Wl,--subsystem,windows,--out-implib,lib/libsass.a
 
 %.o: %.c
 	$(CC) $(CFLAGS) -c -o $@ $<
