--- LPS-53597/JavaServiceDesc.java	2013-04-12 15:11:07.380006000 +0200
+++ src/org/apache/axis/description/JavaServiceDesc.java	2015-02-15 00:06:00.000000000 -0800
@@ -701,10 +707,10 @@
                         }
                         if (paramClass == null) {
                             paramClass = getTypeMapping().getClassForQName(param.getTypeQName(),
-                                                                           type);
+                                                                           null);
                         }
 
-                        if (paramClass != null) {
+                        if (paramClass != null && paramClass != Object.class) {
                             // This is a match if the paramClass is somehow
                             // convertable to the "real" parameter type.  If not,
                             // break out of this loop.
