diff --git a/org/phidias/compile/BundleJavaManager.java b/org/phidias/compile/BundleJavaManager.java
index 46f8d47..302ebab 100644
--- a/org/phidias/compile/BundleJavaManager.java
+++ b/org/phidias/compile/BundleJavaManager.java
@@ -1,3 +1,4 @@
+/* @generated */
 /**
  * Copyright 2012 Liferay Inc.
  *
@@ -222,7 +223,11 @@ public class BundleJavaManager
 	}
 
 	private String getClassNameFromPath(String resourceName) {
-		return resourceName.replace(".class", "").replace("/", ".");
+		if (resourceName.endsWith(".class")) {
+			resourceName = resourceName.substring(0, resourceName.length() - 6);
+		}
+
+		return resourceName.replace('/', '.');
 	}
 
 	private URI getURI(URL url) {
