diff --git a/org/phidias/compile/BundleJavaManager.java b/org/phidias/compile/BundleJavaManager.java
index 44b7206..74052d5 100644
--- a/org/phidias/compile/BundleJavaManager.java
+++ b/org/phidias/compile/BundleJavaManager.java
@@ -230,12 +230,17 @@ public class BundleJavaManager
 		return resourceName.replace('/', '.');
 	}
 
-	private URI getURI(URL url) {
-		String protocol = url.getProtocol();
+	private JavaFileObject getJavaFileObject(
+		URL resourceURL, String resourceName) {
+
+		String protocol = resourceURL.getProtocol();
+
+		String className = getClassNameFromPath(resourceName);
 
 		if (protocol.equals("bundle") || protocol.equals("bundleresource")) {
 			try {
-				return url.toURI();
+				return new BundleJavaFileObject(
+					resourceURL.toURI(), className, resourceURL);
 			}
 			catch (Exception e) {
 				_log.log(e);
@@ -244,52 +249,31 @@ public class BundleJavaManager
 		else if (protocol.equals("jar")) {
 			try {
 				JarURLConnection jarUrlConnection =
-					(JarURLConnection)url.openConnection();
+					(JarURLConnection)resourceURL.openConnection();
 
-				return jarUrlConnection.getJarFileURL().toURI();
+				URL url = jarUrlConnection.getJarFileURL();
+
+				return new JarJavaFileObject(
+					url.toURI(), className, resourceURL, resourceName);
 			}
 			catch (Exception e) {
 				_log.log(e);
 			}
 		}
 		else if (protocol.equals("vfs")) {
-			String file = url.getFile();
-
-			int indexOf = file.indexOf(".jar") + 4;
+			try {
+				String filePath = resourceURL.getFile();
 
-			file =
-				file.substring(0, indexOf) + "!" +
-					url.getFile().substring(indexOf, url.getFile().length());
+				int indexOf = filePath.indexOf(".jar") + 4;
 
-			return new File(file).toURI();
-		}
+				filePath =
+					filePath.substring(0, indexOf) + "!" +
+						filePath.substring(indexOf, filePath.length());
 
-		return null;
-	}
+				File file = new File(filePath);
 
-	private JavaFileObject getJavaFileObject(
-		URL resourceURL, String resourceName) {
+				URI uri = file.toURI();
 
-		String protocol = resourceURL.getProtocol();
-
-		String className = getClassNameFromPath(resourceName);
-
-		URI uri = getURI(resourceURL);
-
-		if (protocol.equals("bundle") || protocol.equals("bundleresource")) {
-			try {
-				return new BundleJavaFileObject(uri, className, resourceURL);
-			}
-			catch (Exception e) {
-				_log.log(e);
-			}
-		}
-		else if (protocol.equals("jar")) {
-			return new JarJavaFileObject(
-				uri, className, resourceURL, resourceName);
-		}
-		else if (protocol.equals("vfs")) {
-			try {
 				return new JarJavaFileObject(
 					uri, className,
 					new URL("jar:" + uri.toString()),
