diff --git a/com/browseengine/bobo/sort/DocComparatorSource.java b/com/browseengine/bobo/sort/DocComparatorSource.java
index 2ad990b..c186b8c 100644
--- a/com/browseengine/bobo/sort/DocComparatorSource.java
+++ b/com/browseengine/bobo/sort/DocComparatorSource.java
@@ -140,6 +140,9 @@ public abstract class DocComparatorSource {
 				}
 
 				public String value(ScoreDoc doc) {
+					if (values.lookup[values.order[doc.doc]] == null) {
+						return null;
+					}
 					return String.valueOf(values.lookup[values.order[doc.doc]]);
 				}
 			};
@@ -338,3 +341,4 @@ public abstract class DocComparatorSource {
 	
 	
 }
+/* @generated */
\ No newline at end of file
diff --git a/com/browseengine/bobo/sort/ReverseDocComparatorSource.java b/com/browseengine/bobo/sort/ReverseDocComparatorSource.java
index 3c203d1..4479743 100644
--- a/com/browseengine/bobo/sort/ReverseDocComparatorSource.java
+++ b/com/browseengine/bobo/sort/ReverseDocComparatorSource.java
@@ -50,6 +50,15 @@ public class ReverseDocComparatorSource extends DocComparatorSource {
 			public int compareTo(Object o) {
 				if (o instanceof ReverseComparable){
 					Comparable inner = ((ReverseComparable)o)._inner;						
+					if (_inner == null) {
+						if (inner == null) {
+							return 0;
+						}
+						return 1;
+					}
+					else if (inner == null) {
+						return -1;
+					}
 					return -_inner.compareTo(inner);
 				}
 				else{
@@ -68,3 +77,4 @@ public class ReverseDocComparatorSource extends DocComparatorSource {
 	}
 
 }
+/* @generated */
\ No newline at end of file
