diff --git a/com/browseengine/bobo/sort/SortCollector.java b/com/browseengine/bobo/sort/SortCollector.java
index 394d9a9..26bf58d 100644
--- a/com/browseengine/bobo/sort/SortCollector.java
+++ b/com/browseengine/bobo/sort/SortCollector.java
@@ -166,7 +166,7 @@ public abstract class SortCollector extends Collector {
 	      return new DocComparatorSource.LongDocComparatorSource(fieldname);
 	
 	    case SortField.DOUBLE:
-	      return new DocComparatorSource.LongDocComparatorSource(fieldname);
+	      return new DocComparatorSource.DoubleDocComparatorSource(fieldname);
 	
 	    case SortField.BYTE:
 	      return new DocComparatorSource.ByteDocComparatorSource(fieldname);
@@ -306,3 +306,4 @@ public abstract class SortCollector extends Collector {
     }
   }
 }
+/* @generated */
\ No newline at end of file
