diff --git a/org/drools/base/ClassFieldAccessorCache.java b/org/drools/base/ClassFieldAccessorCache.java
index 249c48a..cc9e1a5 100644
--- a/org/drools/base/ClassFieldAccessorCache.java
+++ b/org/drools/base/ClassFieldAccessorCache.java
@@ -1,3 +1,4 @@
+/* @generated */
 /*
  * Copyright 2010 JBoss Inc
  *
@@ -123,7 +124,7 @@ public class ClassFieldAccessorCache {
         try {
             return this.classLoader.loadClass( className );
         } catch ( ClassNotFoundException e ) {
-            throw new RuntimeDroolsException( "Unable to resolve class '" + className + "'" );
+            throw new RuntimeDroolsException( "Unable to resolve class '" + className + "'", e );
         }
     }
 
@@ -134,7 +135,7 @@ public class ClassFieldAccessorCache {
         CacheEntry cache = this.cacheByClassLoader.get( cl );
         if ( cache == null ) {
             // setup a cache for this ClassLoader
-            cache = new CacheEntry( this.classLoader );
+            cache = new CacheEntry( com.liferay.portal.kernel.util.AggregateClassLoader.getAggregateClassLoader(new ClassLoader[] {cl, this.classLoader}) );
             this.cacheByClassLoader.put( cl,
                                          cache );
         }
