diff --git a/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradlePomModuleDescriptorBuilder.java b/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradlePomModuleDescriptorBuilder.java
index 123b029..1f60a43 100644
--- a/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradlePomModuleDescriptorBuilder.java
+++ b/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradlePomModuleDescriptorBuilder.java
@@ -183,7 +183,7 @@ public class GradlePomModuleDescriptorBuilder {
             }
         }
 
-        this.mrid = ModuleRevisionId.newInstance(group, module, effectiveVersion);
+        this.mrid = IvyUtil.createModuleRevisionId(group, module, effectiveVersion);
         ivyModuleDescriptor.setModuleRevisionId(mrid);
 
         if (effectiveVersion != null && effectiveVersion.endsWith("SNAPSHOT")) {
@@ -398,4 +398,5 @@ public class GradlePomModuleDescriptorBuilder {
     private PomDependencyMgt findDependencyDefault(PomDependencyData dependency) {
         return pomReader.findDependencyDefaults(dependency.getId());
     }
-}
+}
+/* @generated */
\ No newline at end of file
