diff --git a/org/apache/jasper/runtime/JspWriterImpl.java b/org/apache/jasper/runtime/JspWriterImpl.java
index 9d09f67..3d1b835 100644
--- a/org/apache/jasper/runtime/JspWriterImpl.java
+++ b/org/apache/jasper/runtime/JspWriterImpl.java
@@ -140,7 +140,7 @@ public class JspWriterImpl extends JspWriter {
 	this.response = response;
         allocateCharBuffer();
 
-		if (!response.isCommitted()) {
+		if ((sz > 0) && !response.isCommitted()) {
 			// START OF IASRI 4641975/6172992
 			try {
 				response.setBufferSize(sz);
@@ -157,7 +157,7 @@ public class JspWriterImpl extends JspWriter {
 	this.bufferSize=sz;
         allocateCharBuffer();
 
-		if (!response.isCommitted()) {
+		if ((sz > 0) && !response.isCommitted()) {
 			// START OF IASRI 4641975/6172992
 			try {
 				response.setBufferSize(sz);
