diff --git a/freemarker/ext/beans/BeanModel.java b/freemarker/ext/beans/BeanModel.java
index 40b23fa..6df3b77 100644
--- a/freemarker/ext/beans/BeanModel.java
+++ b/freemarker/ext/beans/BeanModel.java
@@ -227,6 +227,26 @@ implements
     {
         // See if this particular instance has a cached implementation
         // for the requested feature descriptor
+
+        if (desc instanceof PropertyDescriptor) {
+            if (desc instanceof IndexedPropertyDescriptor) {
+                IndexedPropertyDescriptor indexedPropertyDescriptor =
+                    (IndexedPropertyDescriptor)desc;
+
+                desc = indexedPropertyDescriptor.getIndexedReadMethod();
+            }
+            else {
+                PropertyDescriptor propertyDescriptor =
+                    (PropertyDescriptor)desc;
+
+                return wrapper.invokeMethod(
+                    object, propertyDescriptor.getReadMethod(), null);
+            }
+        }
+        else if (desc instanceof Field) {
+            return wrapper.wrap(((Field)desc).get(object));
+        }
+
         TemplateModel member;
         synchronized(this) {
             if(memberMap != null) {
@@ -241,26 +261,7 @@ implements
             return member;
 
         TemplateModel retval = UNKNOWN;
-        if(desc instanceof IndexedPropertyDescriptor)
-        {
-            Method readMethod = 
-                ((IndexedPropertyDescriptor)desc).getIndexedReadMethod(); 
-            retval = member = 
-                new SimpleMethodModel(object, readMethod, 
-                        ClassIntrospector.getArgTypes(classInfo, readMethod), wrapper);
-        }
-        else if(desc instanceof PropertyDescriptor)
-        {
-            PropertyDescriptor pd = (PropertyDescriptor)desc;
-            retval = wrapper.invokeMethod(object, pd.getReadMethod(), null);
-            // (member == null) condition remains, as we don't cache these
-        }
-        else if(desc instanceof Field)
-        {
-            retval = wrapper.wrap(((Field)desc).get(object));
-            // (member == null) condition remains, as we don't cache these
-        }
-        else if(desc instanceof Method)
+        if (desc instanceof Method)
         {
             Method method = (Method)desc;
             retval = member = new SimpleMethodModel(object, method, 
@@ -396,4 +397,4 @@ implements
         return wrapper.wrapAsAPI(object);
     }
     
-}
\ No newline at end of file
+}/* @generated */
\ No newline at end of file
